pkg/driver/controller.go

```diff
@@ -592,16 +594,33 @@ func (cs *controller) CreateSnapshot(
 		)
 	}
 
+	capacity, _ := strconv.ParseInt(vol.Spec.Capacity, 10, 64)
+
+	params, err := NewSnapshotParams(req.GetParameters(), capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"failed to parse csi volume params: %v", err)
+	}
+
+	if !params.AbsSnapSize {
+		snapSize = int64(float64(capacity) * (params.SnapSize / 100))
```

Comment:
can you test the percentage again? it seems like we will always get 0 here.

the calculation should be this :
int64((float64(capacity) * params.SnapSize) / 100)

Reply:
nvm, I see that params.SnapSize is float. So it should be fine.

Reply:
No as the params.SnapSize is float64 (params.SnapSize / 100) will never be 0.

Reply:
I thought this multiplaction (float64(capacity) * params.SnapSize can become a problem if the size of volume is big in TBs

Reply:
@shubham14bajpai Can we have this calculations moved to a separate func() and have unit tests for it.